packages/integrations/cloudflare/src/index.ts

```diff
@@ -228,10 +228,30 @@ export default function createIntegration(args?: Options): AstroIntegration {
 							{
 								name: '@astrojs/cloudflare:environment',
 								configEnvironment(environmentName, _options) {
-									if (environmentName === 'ssr' && _options.optimizeDeps?.noDiscovery === false) {
+									const isServerEnvironment = ['ssr', 'prerender'].includes(environmentName);
+									if (isServerEnvironment && _options.optimizeDeps?.noDiscovery === false) {
 										return {
 											optimizeDeps: {
-												exclude: ['unstorage/drivers/cloudflare-kv-binding'],
+												include: [
```

Comment:
You might want to consider https://github.com/svitejs/vitefu

We already use it in the Astro core to retrieve the dependencies to optimise

Reply:
That's not sufficient, we need to optimize these exact dependencies as well, they are discovered by vitefu.